Makefile

```diff
@@ -92,6 +92,10 @@ endif
 build: ensure-submodule-initialized buf build-module
 	cd viam-cartographer && cmake -Bbuild -G Ninja ${EXTRA_CMAKE_FLAGS} && cmake --build build
 
+build-asan: EXTRA_CMAKE_FLAGS += -DCMAKE_CXXFLAGS="-fno-omit-frame-pointer -fsanitize=address -fsanitize-address-use-after-scope -O1" -DCMAKE_EXE_LINKER_FLAGS="-fsanitize=address"
```

Comment:
I can't add these to build-debug yet as they fail on arm64 linux as that platform currently uses gcc as opposed to clang & gcc doesn't support using asan in this way (more work would be needed to get it to work there).

As as result, I'm going to leave this as an optional build option for now.  Mainly this is intended to be used on osx (which does use clang) where we can't use Valgrind.

Reply:
Mind adding this as a comment in the code?

Reply:
yes